src/user/user.service.ts

```diff
@@ -4,14 +4,71 @@ import { UserRepository } from './user.repository';
 import { InjectRepository } from '@nestjs/typeorm';
 import { formatUserObject } from '../utils/constants';
 import { GetUserDto } from './dto/get-user.dto';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UserEntity } from 'src/entities/user.entity';
+import { PartnerAccessService } from 'src/partner-access/partner-access.service';
+import { PartnerAccessEntity } from 'src/entities/partner-access.entity';
+import { PartnerEntity } from 'src/entities/partner.entity';
+import { PartnerRepository } from 'src/partner/partner.repository';
 
 @Injectable()
 export class UserService {
   constructor(
     @InjectRepository(UserRepository)
     private userRepository: UserRepository,
+    @InjectRepository(PartnerRepository)
+    private partnerRepository: PartnerRepository,
+    private readonly partnerAccessService: PartnerAccessService,
   ) {}
 
+  public async createUser(
+    createUserDto: CreateUserDto,
+  ): Promise<
+    | { user: UserEntity; partnerAccess: PartnerAccessEntity; partner: PartnerEntity }
+    | { user: UserEntity }
+  > {
+    const { name, email, firebaseUid, languageDefault, partnerAccessCode, contactPermission } =
+      createUserDto;
+    const createUserObject = this.userRepository.create({
+      name,
+      email,
+      languageDefault,
+      firebaseUid,
+      contactPermission,
+    });
```

Comment:
what happens if there's an error on this `createUserObject` block above? Is the error handled automatically